bot/storage-mongo/src/main/kotlin/FeatureMongoDAO.kt

```diff
@@ -101,25 +63,52 @@ internal object FeatureMongoDAO : FeatureDAO {
     ): Boolean {
         val id = calculateId(botId, namespace, category, name, applicationId)
         val idWithoutApplicationId = calculateId(botId, namespace, category, name, null)
-        val now = ZonedDateTime.now(internalDefaultZoneId)
 
-        return (features[id] ?: features[idWithoutApplicationId] ?: col.findOne(_id eq id)
-        ?: col.findOne(_id eq idWithoutApplicationId))
-            .let { f ->
-                if (f == null) {
-                    default.also {
-                        addFeature(botId, namespace, default, category, name, null, null)
-                    }
-                } else {
-                    features[id] = f
-                    when {
-                        f.startDate != null && f.endDate == null -> f.enabled && now.isAfter(f.startDate)
-                        f.startDate != null && f.endDate != null -> f.enabled && now.isAfter(f.startDate)
-                            && now.isBefore(f.endDate)
-                        else -> f.enabled
-                    }
-                }
+        val (connectorFeature, globalFeature) = getValues(id, idWithoutApplicationId)
+
+        return if (connectorFeature == null && globalFeature == null) {
+            default.also {
+                addFeature(botId, namespace, default, category, name, null, null, null)
             }
+        } else {
+            val feature = (connectorFeature ?: globalFeature)!!
+            isEnabled(feature)
+        }
+    }
+
+    private fun isEnabled(feature: Feature): Boolean {
+        val now = ZonedDateTime.now(internalDefaultZoneId)
+        return when {
+            feature.startDate != null && feature.endDate == null -> feature.enabled && now.isAfter(feature.startDate)
+            feature.startDate != null && feature.endDate != null -> feature.enabled && now.isAfter(feature.startDate)
+                    && now.isBefore(feature.endDate)
+            else -> feature.enabled
+        }
+    }
+
+    private fun getValues(id: String, idWithoutApplicationId: String): Pair<Feature?, Feature?> {
```

Comment:
Simple object instead of Pair<Feature, Feature> can have methods for :
- connectorFeature == null && globalFeature == null
- (connectorFeature ?: globalFeature)!!